serverless/serverless
# Add service wide metrics support

## What did you implement:

Closes #2846 

Add service wide metrics support via `serverless metrics` command.

## How did you implement it:

Metrics fetching will now be based on an array of functions (e.g. `['func1', 'func2']`) so it defaults to all the service functions. If a `function` option is specified this `functions` array will be filtered and only the specified function will remain in the array.

So the metrics fetching will always be based on an array. It's just a matter of the items which remain in the array.

## How can we verify it:

Deploy a service, invoke some functions and then run:

```bash
serverless metrics
```

to see the service wide metrics.

Additionally run

```bash
serverless metrics --function <functionName>
```

to see all the metrics for the specified function.


## Todos:


***Is this ready for review?:*** YES
